cmd/ci-secret-bootstrap/main.go

```diff
@@ -367,6 +371,12 @@ func constructSecrets(config secretbootstrap.Config, client secrets.ReadOnlyClie
 			// that we need to wait for that map to be fully populated.
 			keyWg.Wait()
 
+			// We don't want to sync secrets that have not been fully fetched from the secret manager
+			// and/or have not been properly constructed.
+			if secretInError.Load() {
+				return
```

Comment:
Maybe an explicit log entry here to say it is being skipped? Nice for someone reading the logs to have a breadcrumb know code was skipped.